Derived from: https://github.com/kiler129/relax-intel-rmrr/blob/master/patches/add-relaxable-rmrr-5_8_and_up.patch
This RMRR patch is for people with older but widespread HP Proliant servers, who need the patch for device passthrough
to VMs.
--- a/drivers/iommu/intel/iommu.c
+++ b/drivers/iommu/intel/iommu.c
@@ -298,6 +298,7 @@
 static int intel_iommu_superpage = 1;
 static int iommu_identity_mapping;
 static int iommu_skip_te_disable;
+static int intel_relaxable_rmrr = 0;
 
 #define IDENTMAP_GFX		2
 #define IDENTMAP_AZALIA		4
@@ -335,7 +336,10 @@
 		return -EINVAL;
 
 	while (*str) {
-		if (!strncmp(str, "on", 2)) {
+		if (!strncmp(str, "relax_rmrr", 10)) {
+			pr_info("Intel-IOMMU: assuming all RMRRs are relaxable. This can lead to instability or data loss\n");
+			intel_relaxable_rmrr = 1;
+		} else if (!strncmp(str, "on", 2)) {
 			dmar_disabled = 0;
 			pr_info("IOMMU enabled\n");
 		} else if (!strncmp(str, "off", 3)) {
@@ -2628,7 +2632,7 @@
 		return false;
 
 	pdev = to_pci_dev(dev);
-	if (IS_USB_DEVICE(pdev) || IS_GFX_DEVICE(pdev))
+	if (intel_relaxable_rmrr || IS_USB_DEVICE(pdev) || IS_GFX_DEVICE(pdev))
 		return true;
 	else
 		return false;
